transport/x/grpc/config.go

```diff
@@ -0,0 +1,126 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package grpc
+
+import (
+	"fmt"
+	"net"
+
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/x/config"
+)
+
+const transportName = "grpc"
+
+// TransportSpec returns a TransportSpec for the gRPC transport.
+//
+// See InboundConfig and OutboundConfig for details on the
+// different configuration parameters supported by this Transport.
+//
+// Any InboundOption or OutboundOption may be passed to this function.
+// These options will be applied BEFORE configuration parameters are
+// interpreted. This allows configuration parameters to override Options
+// provided to TransportSpec.
+func TransportSpec(opts ...Option) config.TransportSpec {
+	transportSpec, err := newTransportSpec(opts...)
+	if err != nil {
+		panic(err.Error())
+	}
+	return config.TransportSpec{
+		Name:               transportName,
+		BuildTransport:     transportSpec.buildTransport,
+		BuildInbound:       transportSpec.buildInbound,
+		BuildUnaryOutbound: transportSpec.buildUnaryOutbound,
+	}
+}
+
+// InboundConfig configures a gRPC Inbound.
+//
+// inbounds:
+//   grpc:
+//     address: ":80
+type InboundConfig struct {
+	// Address to listen on. This field is required.
+	Address string `config:"address,interpolate"`
+}
+
+// OutboundConfig configures a gRPC Outbound.
+//
+// outbounds:
+//   myservice:
+//     grpc:
+//       address: ":80
+type OutboundConfig struct {
+	// Address to connect to. This field is required.
+	Address string `config:"address,interpolate"`
+}
+
+type transportSpec struct {
+	InboundOptions  []InboundOption
+	OutboundOptions []OutboundOption
+}
+
+func newTransportSpec(opts ...Option) (*transportSpec, error) {
+	transportSpec := &transportSpec{}
+	for _, o := range opts {
+		switch opt := o.(type) {
+		case InboundOption:
+			transportSpec.InboundOptions = append(transportSpec.InboundOptions, opt)
+		case OutboundOption:
+			transportSpec.OutboundOptions = append(transportSpec.OutboundOptions, opt)
+		default:
+			return nil, fmt.Errorf("unknown option of type %T: %v", o, o)
+		}
+	}
+	return transportSpec, nil
+}
+
+func (t *transportSpec) buildTransport(_ struct{}, _ *config.Kit) (transport.Transport, error) {
```

Comment:
nit: if all parameters are `_`, you can skip the `_`s completely.

    buildTransport(struct{}, *config.Kit) (transport.Transport, error) { ... }